pickapp/src/main/kotlin/com/lovoo/android/pickapp/adapter/SelectionAdapter.kt

```diff
@@ -108,40 +111,19 @@ class SelectionAdapter : RecyclerView.Adapter<SelectionAdapter.ViewHolder>() {
         view: View,
         private val imageEngine: ImageEngine
     ) : RecyclerView.ViewHolder(view) {
-
-        val size = MutableLiveData<Int>()
-
-        init {
-            view.viewTreeObserver.addOnPreDrawListener(object : ViewTreeObserver.OnPreDrawListener {
-                override fun onPreDraw(): Boolean {
-                    view.viewTreeObserver.removeOnPreDrawListener(this)
-                    size.postValue(view.measuredWidth)
-                    return true
-                }
-            })
-        }
+        private val width = view.context.resources.getDimensionPixelSize(R.dimen.pickpic_thumbnail_size)
+        private val corner = view.context.resources.getDimensionPixelSize(R.dimen.pickpic_thumbnail_corner_size)
 
         fun bind(uri: Uri?, isSelected: Boolean, onClickListener: ((View, Uri) -> Unit)?) {
             itemView.isSelected = isSelected
             itemView.setOnClickListener { view ->
                 uri?.let { onClickListener?.invoke(view, it) }
             }
-            (itemView as? ImageView)?.setImageBitmap(null)
+            val imageView = itemView.findViewById<ImageView>(R.id.pickpic_thumbnail_image)
+            imageView?.setImageBitmap(null)
             if (uri == null) return
-
-            val width = size.value ?: 0
-            if (width == 0) {
-                size.observeForever(object : Observer<Int> {
-                    override fun onChanged(width: Int?) {
-                        size.removeObserver(this)
-                        bind(uri, isSelected, onClickListener)
-                    }
-                })
-                return
-            }
-
-            (itemView as? ImageView)?.let {
-                imageEngine.loadThumbnail(it.context, width, uri, it, 0)
+            imageView?.let {
+                imageEngine.loadThumbnail(it.context, width, uri, it, corner)
             }
         }
```

Comment:
Just a suggestion, to avoid val, ? and ?let

```
itemView.findViewById<ImageView>(R.id.pickpic_thumbnail_image)?.apply {
   setImageBitmap(null)
   if (uri != null)  imageEngine.loadThumbnail(...)
}
```

Reply:
@Nidvex you can submit code change suggestions : https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request

Step 6
you can also mark multiple lines when you press the +/- line and drag down

Reply:
yeah I wanted to do this. doesn't work if there are <deleted> lines in between :D